sweep/sweeper.go

```diff
@@ -880,6 +909,45 @@ func (s *UtxoSweeper) waitForSpend(outpoint wire.OutPoint,
 	return spendEvent.Cancel, nil
 }
 
+// PendingInputs returns the set of inputs that the UtxoSweeper is currently
+// attempting to sweep.
+func (s *UtxoSweeper) PendingInputs() (pendingInputs, error) {
+	respChan := make(chan pendingInputs, 1)
+	select {
+	case s.pendingSweepsReqs <- &pendingSweepsReq{
+		respChan: respChan,
+	}:
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+
+	select {
+	case pendingSweeps := <-respChan:
+		return pendingSweeps, nil
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+}
+
+// handlePendingSweepsReq handles a request to retrieve all pending inputs the
+// UtxoSweeper is attempting to sweep.
+func (s *UtxoSweeper) handlePendingSweepsReq(req *pendingSweepsReq) pendingInputs {
+	pendingInputs := make(pendingInputs, len(s.pendingInputs))
+	for _, pendingInput := range s.pendingInputs {
+		// Only the exported fields are set, as we expect the response
+		// to only be consumed externally.
+		pendingInputs[*pendingInput.OutPoint()] = &PendingInput{
```

Comment:
I still feel that usage of `PendingInput` and the embedded `input.Input` to expose the sweeper state is not good from an interface segregation perspective. A new struct may just be 10 lines of code and will take away any doubts about what is and isn't populated / meant to be read.

Reply:
Done.

Reply:
I think it looks better